src/unit_test.c

```diff
@@ -0,0 +1,407 @@
+/***********************************************************************
+ * Copyright (c) 2025  Matias Furszyfer (furszy)                       *
+ * Distributed under the MIT software license, see the accompanying    *
+ * file COPYING or https://www.opensource.org/licenses/mit-license.php.*
+ ***********************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* Check for POSIX process headers */
+#if defined(__has_include)
+#  if __has_include(<sys/wait.h>) && __has_include(<unistd.h>)
+#    include <sys/wait.h>
+#    include <unistd.h>
+/* Ensure pid_t is defined. Include <sys/types.h> only if necessary (strict C89 mode) */
+#if !defined(_PID_T) && !defined(pid_t)
+#   include <sys/types.h>
+#endif
+#    define SUPPORTS_CONCURRENCY 1
+#  endif
+#endif
```

Comment:
I'm not sure about this implementation.

1. `__has_include` works fine with modern compilers. However, it is a [C23 feature](https://en.cppreference.com/w/c/preprocessor/include.html). Should we rely on it?

2. The availability of headers and symbols is usually a task for the build system, which defines `SUPPORTS_CONCURRENCY` based on the header / symbol checks.

Reply:
> I'm not sure about this implementation:
> 
> 1. `__has_include` works fine with modern compilers. However, it is a [C23 feature](https://en.cppreference.com/w/c/preprocessor/include.html). Should we rely on it?
> 2. The availability of headers and symbols is usually a task for the build system, which defines `SUPPORTS_CONCURRENCY` based on the header / symbol checks.

Hmm, yeah. I was trying to avoid touching the build systems (since we have two now), but can move the code there too. The guards here are mainly to postpone Windows support for future work (if we ever want to add support for such platform).

Reply:
Done. Let me know if the build system changes can be shrunk even more. I tried to keep them minimal.